presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -539,7 +550,13 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
         configBinder(binder).bindConfig(StaticFunctionNamespaceStoreConfig.class);
         binder.bind(FunctionAndTypeManager.class).in(Scopes.SINGLETON);
         binder.bind(MetadataManager.class).in(Scopes.SINGLETON);
-        binder.bind(Metadata.class).to(MetadataManager.class).in(Scopes.SINGLETON);
+        if (serverConfig.isCatalogServerEnabled() && serverConfig.isCoordinator()) {
+            binder.bind(RemoteMetadataManager.class).in(Scopes.SINGLETON);
```

Comment:
This should never be True, right? A server can not be both coordinator and catalogServer?

Reply:
No, isCatalogServerEnabled does not mean that the server is a catalogServer, that is isCatalogServer. What this check does is first see if the catalogServer is enabled at all meaning we are planning on actually using the catalog server and then if so the coordinator will make use of the remoteMetadataManager which is needed to call the catalog server instead of the standard one